lib/ansible/modules/cloud/vmware/vmware_guest_register_operation.py

```diff
@@ -0,0 +1,266 @@
+            }
+            self.module.fail_json(msg="No folder %s matched in the search path : %s" % (self.folder, fullpath),
+                                  details=details)
+
+        if self.state == "present":
+            if self.get_vm():
+                self.module.exit_json(**result)
+            else:
+                if self.esxi_hostname:
+                    host_obj = self.find_hostsystem_by_name(self.esxi_hostname)
+                    if not host_obj:
+                        self.module.fail_json(msg="Cannot find the specified ESXi host: %s" % self.esxi_hostname)
+                else:
+                    host_obj = None
+
+                if self.cluster_name:
+                    cluster_obj = find_cluster_by_name(self.content, self.cluster_name, datacenter)
+                    if not cluster_obj:
+                        self.module.fail_json(msg="Cannot find the specified cluster name: %s" % self.cluster_name)
+
+                    resource_pool_obj = cluster_obj.resourcePool
+                elif self.resource_pool:
+                    resource_pool_obj = find_resource_pool_by_name(self.content, self.resource_pool)
+                    if not resource_pool_obj:
+                        self.module.fail_json(msg="Cannot find the specified resource pool: %s" % self.resource_pool)
+                else:
+                    resource_pool_obj = host_obj.parent.resourcePool
+
+                task = folder_obj.RegisterVM_Task(path=self.path, name=self.name, asTemplate=self.template,
+                                                  pool=resource_pool_obj, host=host_obj)
+
+                changed = False
+                try:
+                    changed, info = wait_for_task(task)
+                except Exception as task_e:
+                    self.module.fail_json(msg=to_native(task_e))
+
+                result.update(changed=changed)
+                self.module.exit_json(**result)
+
+        else:
+            vm_obj = self.get_vm()
+            if vm_obj:
+                try:
+                    vm_obj.UnregisterVM()
+                    result.update(changed=True)
+                except Exception as exc:
+                    self.module.fail_json(msg=to_native(exc))
+
+            self.module.exit_json(**result)
+
+
+def main():
+    argument_spec = vmware_argument_spec()
+    argument_spec.update(datacenter=dict(type="str", default="ha-datacenter"),
+                         cluster_name=dict(type="str"),
+                         folder=dict(type="str"),
+                         name=dict(type="str", required=True),
+                         uuid=dict(type="str"),
+                         esxi_hostname=dict(type="str"),
+                         path=dict(type="str"),
+                         template=dict(type="bool", default=False),
+                         resource_pool=dict(type="str"),
+                         state=dict(type="str", default="present", cohices=["present", "absent"]))
```

Comment:
```suggestion
                         state=dict(type="str", default="present", choices=["present", "absent"]))
```

Reply:
Thank you for pointing it out.
Fixed typo.